snuba/utils/kafka/consumers/abstract.py

```diff
@@ -0,0 +1,30 @@
+from abc import ABC, abstractmethod
+from typing import Any, Callable, Generic, Optional, Sequence, TypeVar
+
+from confluent_kafka import TopicPartition
+
+
+TMessage = TypeVar("TMessage")
+
+
+class Consumer(Generic[TMessage], ABC):
+    @abstractmethod
+    def subscribe(
+        self,
+        topics: Sequence[str],
+        on_assign: Optional[Callable[[Sequence[TopicPartition]], Any]] = None,
+        on_revoke: Optional[Callable[[Sequence[TopicPartition]], Any]] = None,
```

Comment:
These differ from the default assignment/revocation callbacks that have the type `Optional[Callable[[Consumer, Sequence[TopicPartition]], Any]]`.

The consumer seems redundant here — if you need the consumer, just partial it to your callback — so I removed it.